### ISSUE TITLE
asserts modifier not supported

### ISSUE THREAD

**Issue by unional** [typescript-eslint/typescript-eslint#1218]
**What code were you trying to parse?**

```ts
function foo(v: any): asserts v is string {
}
```

**What did you expect to happen?**
Support the asserts modifier

**What actually happened?**
![image](https://user-images.githubusercontent.com/3254987/69018106-8d933780-095f-11ea-8a31-d3e1f56100ff.png)

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `2.7.0` |
| `TypeScript`                | `3.7.2` |
| `ESLint`                    | `6.6.0` |
| `node`                      | `12.8.1` |
| `npm`                       | `6.13.0` |

**Comment by bradzacher** [comment]
I'm unable to repro this against master.

There are passing tests in master for this.
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/shared-fixtures/fixtures/typescript/basics/type-assertion-with-guard-in-function.src.ts

Could you please provide some more information?

**Comment by unional** [comment]
Not sure why it doesn't work then.
I check `yarn.lock` and confirm I'm using 2.7.0
```
"@typescript-eslint/parser@^2.5.0", "@typescript-eslint/parser@^2.7.0":
  version "2.7.0"
  resolved "https://registry.yarnpkg.com/@typescript-eslint/parser/-/parser-2.7.0.tgz#b5e6a4944e2b68dba1e7fbfd5242e09ff552fd12"
  integrity sha512-ctC0g0ZvYclxMh/xI+tyqP0EC2fAo6KicN9Wm2EIao+8OppLfxji7KAGJosQHSGBj3TcqUrA96AjgXuKa5ob2g==
  dependencies:
    "@types/eslint-visitor-keys" "^1.0.0"
    "@typescript-eslint/experimental-utils" "2.7.0"
    "@typescript-eslint/typescript-estree" "2.7.0"
    eslint-visitor-keys "^1.1.0"
```

**Comment by bradzacher** [comment]
Are you using prettier? You have to update to the latest or else it will fail.
Restarted your IDE?
Deleted and reinstalled node modules?
I'm not sure what the problem might be.

I can't fix or look into anything here without an isolated repro, esp when it's working fine in master.
Feel free to comment back if you have a repro, and I can reopen to investigate.

**Comment by unional** [comment]
It starts to work after I restart VSCode. :)
